packages/relay/src/lib/eth.ts

```diff
@@ -1187,29 +1183,34 @@ export class EthImpl implements Eth {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     return this.mirrorNodeClient.getContractResultsByAddressAndTimestamp(to, timestamp, requestId)
       .then(contractResultDetails => {
-        const rSig = contractResultDetails.r === null ? null : contractResultDetails.r.substring(0, 66);
-        const sSig = contractResultDetails.s === null ? null : contractResultDetails.s.substring(0, 66);
-        return new Transaction({
-          accessList: undefined, // we don't support access lists for now, so punt
-          blockHash: contractResultDetails.block_hash.substring(0, 66),
-          blockNumber: EthImpl.numberTo0x(contractResultDetails.block_number),
-          chainId: contractResultDetails.chain_id,
-          from: contractResultDetails.from.substring(0, 42),
-          gas: EthImpl.nanOrNumberTo0x(contractResultDetails.gas_used),
-          gasPrice: EthImpl.toNullIfEmptyHex(contractResultDetails.gas_price),
-          hash: contractResultDetails.hash.substring(0, 66),
-          input: contractResultDetails.function_parameters,
-          maxPriorityFeePerGas: EthImpl.toNullIfEmptyHex(contractResultDetails.max_priority_fee_per_gas),
-          maxFeePerGas: EthImpl.toNullIfEmptyHex(contractResultDetails.max_fee_per_gas),
-          nonce: EthImpl.nanOrNumberTo0x(contractResultDetails.nonce),
-          r: rSig,
-          s: sSig,
-          to: contractResultDetails.to.substring(0, 42),
-          transactionIndex: EthImpl.numberTo0x(contractResultDetails.transaction_index),
-          type: EthImpl.nullableNumberTo0x(contractResultDetails.type),
-          v: EthImpl.nullableNumberTo0x(contractResultDetails.v),
-          value: EthImpl.numberTo0x(contractResultDetails.amount),
-        });
+        // 404 is allowed return code so it's possible for contractResultDetails to be null
+        if(contractResultDetails == null) {
+          return null;
```

Comment:
Doe eth.spec.ts have this coverage already or does it need to be added?

Reply:
I think it was, but I'll double-check.

I found something else in for that catch, that may be a potential issue. Currently we are catching any errors and returning `null` for that transaction. For example if a `contractResultDetails` property is `null` and some operation is done on it, the code will break with an error. For example if the block_hash here https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1194 is null for some reason (which is an unexpected behaviour and we would like to be aware if it happens) the error will be caught by the `catch` block and it will just return null. I've tested it in the `openrpc` specs and if I set the block_hash to `null` in the mock here https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/tests/lib/openrpc.spec.ts#L124, the `eth_getBlockByNumber` tests will still return a response, but with one transaction missing. 

Usually it's a good practice in the catch blocks to catch specific error or behaviours that are expected to happen, but let the rest throw an error and break, so we are aware that there's an unexpected issue or behaviour. While currently it will log the error to the console, it seems the response still succeeds, and the issue can easily be missed.

Reply:
In regards with the tests, mocks for `getContractResultsByAddressAndTimestamp` were missing for the `eth_getBlockByNumber`, which resulted in a `no mock` error, which was handled by the `catch` block and the `getTransactionFromContractResult` function returned `null`. Currently I've set all mocks to return 404, because the specs only test that there's a result, aka a block is returned, and that the block number is the expected one.

Reply:
Agreed. We should catch expected errors and throw unexpected so we fix them.

Reply:
You can update this PR or open a ticket highlighting the improved logic